packages/api/tests/endpoints/providers/create-provider.js

```diff
@@ -196,3 +196,26 @@ test('POST returns a 400 response if invalid hostname is provided', async (t) =>
     .expect(400);
   t.is(response.status, 400);
 });
+
+test('CUMULUS-176 POST returns a 404 if the requested path does not exist', async (t) => {
+  const newProvider = fakeProviderFactory();
+
+  const response = await request(app)
+    .post(`/providers/${newProvider.providerid}`)
+    .send(newProvider)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  t.is(response.statusCode, 404);
+});
+
+test('CUMULUS-176 POST returns a 400 response if invalid JSON provided', async (t) => {
+  const response = await request(app)
+    .post('/providers')
+    .set('Content-Type', 'application/json')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send('asdf');
+
+  t.is(response.statusCode, 400);
```

Comment:
Would it be good to test the response message if invalid JSON is provided, since the intent here seems to be improving clarity?

Reply:
Fixed here: dc1e36b7208a834cc10c51e675d47c41859820a8